autorest/utility_test.go

```diff
@@ -234,6 +234,29 @@ func TestStringWithValidString(t *testing.T) {
 	}
 }
 
+func TestStringWithStringSlice(t *testing.T) {
+	s := []string{"string1", "string2"}
+	if String(s, ",") != "string1,string2" {
+		t.Fatal("autorest: String method failed to convert string slice to string")
```

Comment:
[nit] the more go-ish way of expressing this would be to do something like the following:

``` Go
func TestStringWithStringSlice(t *testing.T) {
    s := []string{"string1", "string2"}
    if got, want := String(s, ","), "string1,string2"; got != want {
        t.Logf("got:  %q\nwant: %q", got, want)
        t.Fail()
    }
}
```